Compiler/FunctionCompiler.py

```diff
@@ -785,7 +785,30 @@ def compile_while(self): #  while (expression) statement       note - statement
         code += inner_scope_code  # <while> scope code. after this code, pointer points to the next available cell. i.e one after the expression
         code += expression_code  # re-evaluate the expression
         code += "<"  # point to the expression
-        code += "]"  # after <if> scope
+        code += "]"  # after <while> scope
+
+        return code
+
+    def compile_do_while(self):  # do statement while (expression)       note - statement can be scope { }
+        self.parser.check_current_tokens_are([Token.DO])
+        self.parser.advance_token()
+
+        inner_scope_code = self.compile_statement()
+
+        self.parser.check_current_tokens_are([Token.WHILE, Token.LPAREN])
+        self.parser.advance_token(amount=2)  # point to after LPAREN
+
+        expression_code = self.compile_expression()
+
+        self.parser.check_current_tokens_are([Token.RPAREN, Token.SEMICOLON])
+        self.parser.advance_token()  # point to after RPAREN
+
+        code = "[-]+"  # set expression to 1. since do while loops executes the scope code first.
+        code += "["  # go in scope
+        code += inner_scope_code  # <do-while> scope code. after this code, pointer points to the next available cell. i.e one after the expression
```

Comment:
<do-while> scope code. after this code, pointer points to the same cell. i.e the expression